src/python/pants/backend/python/dependency_inference/module_mapper.py

```diff
@@ -57,16 +79,28 @@ def addresses_for_module(self, module: str) -> Tuple[Address, ...]:
         # imports, where we don't care about the specific symbol, but only the module. For example,
         # with `from my_project.app import App`, we only care about the `my_project.app` part.
         #
-        # We do not look past the direct parent, as this could cause multiple ambiguous owners to be resolved. This
-        # contrasts with the third-party module mapping, which will try every ancestor.
+        # We do not look past the direct parent, as this could cause multiple ambiguous owners to
+        # be resolved. This contrasts with the third-party module mapping, which will try every
+        # ancestor.
         if "." not in module:
             return ()
         parent_module = module.rsplit(".", maxsplit=1)[0]
         return self.mapping.get(parent_module, ())
 
 
 @rule(desc="Creating map of first party targets to Python modules", level=LogLevel.DEBUG)
-async def map_first_party_modules_to_addresses() -> FirstPartyModuleToAddressMapping:
+async def map_first_party_modules_to_addresses(
+    union_membership: UnionMembership,
+) -> PythonFirstPartyModuleMapping:
+    plugin_mappings = await MultiGet(
+        Get(
+            PythonFirstPartyModuleMappingPlugin,
+            PythonFirstPartyModuleMappingPluginRequest,
+            request_cls(),
+        )
+        for request_cls in union_membership.get(PythonFirstPartyModuleMappingPluginRequest)
+    )
```

Comment:
This won't run in parallel with the rest of the body of this rule. It's not critical, but another factoring might be to have both the existing and new providers of `PythonFirstPartyModuleMapping` be implemented uniformly behind the union, which would result in all of them running in parallel in this first multiget.

Reply:
Hm, good point. This rule gets invoked so frequently and it's costly to run that I think this is worth fixing.